src/main/java/de/tum/in/www1/artemis/web/rest/ExamResource.java

```diff
@@ -104,18 +94,71 @@ public ExamResource(UserService userService, CourseService courseService, ExamSe
             return createExam(courseId, updatedExam);
         }
 
-        User user = userService.getUserWithGroupsAndAuthorities();
-        Course course = courseService.findOne(courseId);
-        if (!authCheckService.isAtLeastInstructorInCourse(course, user)) {
-            return forbidden();
-        }
-
-        Optional<Exam> existingExam = examRepository.findById(updatedExam.getId());
-        if (existingExam.isEmpty()) {
-            return ResponseEntity.notFound().build();
+        Optional<ResponseEntity<Exam>> courseAndExamAccessFailure = examAccessService.checkCourseAndExamAccess(courseId, updatedExam.getId());
+        if (courseAndExamAccessFailure.isPresent()) {
+            return courseAndExamAccessFailure.get();
         }
 
         Exam result = examService.save(updatedExam);
         return ResponseEntity.ok().headers(HeaderUtil.createEntityUpdateAlert(applicationName, true, ENTITY_NAME, result.getTitle())).body(result);
     }
+
+    /**
+     * GET /courses/{courseId}/exams/{examId} : Find an exam by id.
+     *
+     * @param courseId  the course to which the exam belongs
+     * @param examId    the exam to find
+     * @return the ResponseEntity with status 200 (OK) and with the found exam as body
+     */
+    @GetMapping("/courses/{courseId}/exams/{examId}")
+    @PreAuthorize("hasAnyRole('ADMIN', 'INSTRUCTOR')")
+    public ResponseEntity<Exam> getExam(@PathVariable Long courseId, @PathVariable Long examId) {
+        log.debug("REST request to get exam : {}", examId);
+        Optional<ResponseEntity<Exam>> courseAndExamAccessFailure = examAccessService.checkCourseAndExamAccess(courseId, examId);
+        Exam exam = examService.findOne(examId);
+        return courseAndExamAccessFailure.orElseGet(() -> ResponseEntity.ok(exam));
```

Comment:
Can't this be simplified?
```suggestion
        return courseAndExamAccessFailure.orElseGet(() -> ResponseEntity.ok(examService.findOne(examId)));
```